src/app/pages/server-details/server-details.component.ts

```diff
@@ -1159,4 +1167,41 @@ export class ServerDetailsComponent implements OnInit {
       return this.serverDetails.benchmark_scores?.find((b) => b.benchmark_id === 'stress_ng:cpu_all' && (b.config as any)?.cores !== 1)?.score?.toFixed(0) || '-';
     }
   }
+
+  generateSchemaJSON(title: string, description: string, keywords: string) {
+    if(!this.serverDetails) {
+      return;
+    }
+
+    console.log('Generating JSON-LD schema for server details page', this.serverDetails);
+
+    const json: any =
+    // generate schema.org Product JSON-LD
+    //const server = this.serverDetails;
+    {
+      "@context": "https://schema.org/",
+      "@type": "Product",
+      "name": this.serverDetails.display_name,
+      "description": description,
+      "brand": {
+        "@type": "Organization",
+        "name": this.serverDetails.vendor.name,
+        "logo": this.serverDetails.vendor.logo,
+        "url": this.serverDetails.vendor.homepage
+      }
+    }
+
+    if(this.similarByFamily.length) {
+      json['isSimilarTo'] = this.similarByPerformance.map((s) => {
```

Comment:
length check is done for `similarByFamily` but using `similarByPerformance` looks odd

Reply:
oops sry